### PR TITLE
Google Analytics

### PR DESCRIPTION
**Pull request by peter-budo**
@karllindmark build http://ci.peterscorner.co.uk/job/BF4-Intel-PR/33/

 • Added depended library, not available on Maven repo therefore imported from `libs` folder
 • Added analytics.xml
 • Added tracking to BaseFragment and BaseListFragment
 • Changed off screen limit for unlocks to 1(on unlock it was 2 and once activity open it would report usage of all 3 fragments, so to add little more accuracy I moved to 1 off screen as on SoldierStats)
 • Renamed AboutActivity to AppInfoActivity
 • Renamed AppInfoFragment to AboutFragment (hope this will make sense)

If you want to test locally what fragments you are on just add `Log.e("Analytics", this.getClass().getSimpleName());` to BaseFragment and BaseListFragment that have onStart method implementations.

### CODE REVIEW

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/unlocks/UnlockTabFragment.java [hunk 1]
```diff
@@ -42,6 +42,6 @@ public static UnlockTabFragment newInstance(final Bundle data) {
 
     @Override
     protected int getOffscreenPageLimit() {
-        return 2;
+        return 1;
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/unlocks/UnlockTabFragment.java]: ...this change really doesn't make sense.

> Changed off screen limit for unlocks to 1(on unlock it was 2 and once activity open it would report usage of all 3 fragments, so to add little more accuracy I moved to 1 off screen as on SoldierStats)

Are you really considering to give the user a WORSE experience, just so that we can get a bit more accurate analytics? I for one would rather get analytics with a known misrepresentation (all three fragments) than a random misreprentation.

Example in your current implementation:
1. User opens unlocks
   
   1.1. Fragment A and B are downloaded (A=1,B=1)
2. User scrolls to fragment B
   
   2.1. Fragment C is downloaded (A=1, B=1, C=1)
3. User scrolls to fragment C
   
   3.1. Fragment A is cleared (A=1, B=1, C=1)
4. User scrolls to fragment B
   
   4.1.   Fragment A is downloaded (A=2, B=1, C=1)
5. User scrolls to fragment C because he forgot something
   
   5.1.     Fragment A is unloaded (A=2, B=1, C=1)
6. User scrolls to fragment B again
   
   6.1.    Fragment A is downloaded again, third time now (A=3, B=1, C=1)

So, basically, what that would generate would be... three hits for A, one for B and one for C. How is that more accurate? If we load all fragments at the same time, we get one for each, which is a more viable representation.

I'm just trying to make sure that we don't do stuff that makes the app worse for the users (and the experience). For example, I would rather look into if we can use the ViewPager "hooks" and send a "trigger" once we switch to a given fragment. That way, if we go from A -> B -> C -> B -> C, we'll instead notice that the user used fragment B and C twice, and A once. This seems both possible and more useful to me, instead of causing people to re-download information just because we're doing it wrong. :+1: 

@peter-budo

**Comment by peter-budo** [comment]
Gone work on some changes
